pkg/apiserver/rest/dualwriter.go

```diff
@@ -209,7 +226,17 @@ func SetDualWritingMode(
 
 	// 	#TODO add support for other combinations of desired and current modes
 
-	return NewDualWriter(currentMode, legacy, storage, reg), nil
+	options.Mode = currentMode
+	dualWriter := NewDualWriter(legacy, storage, options)
+
+	if currentMode == Mode2 {
```

Comment:
As discussed, we'll need some way of indicating when syncing is complete and we are ready to move to move 3.